lib/net/pool.js

```diff
@@ -4564,6 +4564,20 @@ class PoolOptions {
       this.agent = options.agent;
     }
 
+    if (options.agentComment) {
+      assert(typeof options.agentComment === 'string');
+
+      let agentComment;
+      if (this.agent.endsWith('/')) {
+        agentComment = `${this.agent.slice(0, -1)}(${options.agentComment})/`;
+      } else {
+        agentComment = `${this.agent}(${options.agentComment})`;
+      }
+;
+      assert(agentComment.length <= 255);
+      this.agent = agentComment;
+    }
+
```

Comment:
Many thoughts:
- assertion errors should report why they throw ("...must be string")
- comment should be appended to current (default) agent: `/hsd:v3.0.1/sinpapeles`
- total string should be checked for length and asserted with explanation if error
- why do we care if the ua comment ends with a slash>

Reply:
This implementation follows @nodech's comment:

> `./hsd --agent-comment other-variation` - giving us version - `/hsd:3.0.1(other-variation)/`.